pkg/util/converter/prom.go

```diff
@@ -858,8 +872,9 @@ func readStream(iter *jsoniter.Iterator) backend.DataResponse {
 		return backend.DataResponse{Error: err}
 	}
 
-	for iter.ReadArray() {
-		if iterError() {
+	for more, err := iter.ReadArray(); more; more, err = iter.ReadArray() {
+		if err != nil {
+			rsp.Error = err
```

Comment:
can make this line and next line into one line with utility like `return rspErr(err)`